### PR TITLE
[#4412]  feat(api): add validate check for default position when updating column position

### PR DESCRIPTION
**Pull request by LiuQhahah**
### What changes were proposed in this pull request?

add validate check for ColumnPosition when updating column positions, 
If the position is default, will throw exception.

### Why are the changes needed?

Fix: #4412 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
the UT has been added.

### CODE REVIEW

File: api/src/main/java/org/apache/gravitino/rel/TableChange.java [hunk 1]
```diff
@@ -1353,6 +1353,9 @@ final class UpdateColumnPosition implements ColumnChange {
     private final ColumnPosition position;
 
     private UpdateColumnPosition(String[] fieldName, ColumnPosition position) {
+      if (position instanceof Default){
```
Comment (FANNG1) [inline api/src/main/java/org/apache/gravitino/rel/TableChange.java:1356]: please use Preconditions.checkArgument

**Comment by FANNG1** [comment]
@LiuQhahah , please fix the ci and test failures like `CatalogIcebergBaseIT#testUpdateIcebergColumnDefaultPosition`

File: api/src/main/java/org/apache/gravitino/rel/TableChange.java (hunk 1 above)
Comment (LiuQhahah) [reply api/src/main/java/org/apache/gravitino/rel/TableChange.java]: Hi @FANNG1 
I have use Preconditions.checkArgument for hanlding the argument. 
Thanks
Qiang

**Comment by LiuQhahah** [comment]
I have fixed the testUpdateIcebergColumnDefaultPosition UT and formated the code.

File: catalogs/catalog-lakehouse-iceberg/src/test/java/org/apache/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergBaseIT.java [hunk 2]
```diff
@@ -613,7 +613,7 @@ public void testUpdateIcebergColumnDefaultPosition() {
         .alterTable(
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/test/java/org/apache/gravitino/catalog/lakehouse/iceberg/integration/test/CatalogIcebergBaseIT.java:613]: please remove the test

**Comment by jerryshao** [comment]
Looks like there's some style issue needs to be fixed, @LiuQhahah would you please fix it.

File: api/src/test/java/org/apache/gravitino/TestTableChange.java [hunk 3]
```diff
@@ -168,6 +168,17 @@ public void testUpdateNestedColumnPosition() {
     assertEquals(newPosition, updateColumnPosition.getPosition());
   }
 
+  @Test
+  public void testDoesNotAllowDefaultInColumnPosition() {
+    String[] fieldName = {"Name", "Last Name"};
+    ColumnPosition newPosition = TableChange.ColumnPosition.defaultPos();
+    try {
```
Comment (FANNG1) [inline api/src/test/java/org/apache/gravitino/TestTableChange.java:175]: please use `Assertions.assertThrowsExactly(xxx)` to do the check

**Comment by FANNG1** [comment]
There're some residual logic to process DefaultColumnPosition in `IcebergTableOpsHelper.java`, do you like to remove it in this PR?
```java 
  private void doUpdateColumnPosition(
      UpdateSchema icebergUpdateSchema,
      UpdateColumnPosition updateColumnPosition,
      Schema icebergTableSchema) {
    StructType tableSchema = icebergTableSchema.asStruct();
    // not need any more
    ColumnPosition columnPosition =
        getColumnPositionForIceberg(tableSchema, updateColumnPosition.getPosition());
    doMoveColumn(icebergUpdateSchema, updateColumnPosition.fieldName(), columnPosition);
  }
```

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@LiuQhahah , thanks for your contribution, and hope you enjoy the journey.

**Pull request merged**